Function object, NFE (#3360)
javascript-tutorial/en.javascript.info: 1-js/06-advanced-functions/06-function-object/article.md

As we already know, a function in JavaScript is a value.

> Every value in JavaScript has a type. What is a functions type?

Reviewer: would recommend the below
```suggestion
Every value in JavaScript has a type. What is the type of a function?
```
<br>

if we are keeping the current one, then `functions` should be changed to `function's`

There's no magic though. There are cases when there's no way to figure out the right name. In that case, the name property is empty, like here:

```js run

> // function created inside of an array

Reviewer: would recommend removing the 'of'
```suggestion
// function created inside an array
```

That happens because the function takes `sayHi` from its outer lexical environment. There's no local `sayHi`, so the outer variable is used. And at the moment of the call that outer `sayHi` is `null`.

> The optional name which we can add into the Function Expression is meant to solve exactly these kinds of problems.

Reviewer: would recommend using 'in'
```suggestion
The optional name which we can add in the Function Expression is meant to solve exactly these kinds of problems.
```

Also, functions may carry additional properties. Many well-known JavaScript libraries make great use of this feature.

> They create a "main" function and attach many other "helper" functions to it. For instance, the [jQuery](https://jquery.com) library creates a function named `$`. The [lodash](https://lodash.com) library creates a function `_`, and then adds `_.clone`, `_.keyBy` and other properties to it (see the [Lodash docs](https://lodash.com/docs) when you want to learn more about them). Actually, they do it to lessen their pollution of the global space, so that a single library gives only one global variable. That reduces the possibility of naming conflicts.

Reviewer: can we use consistent casing for lodash in both [lodash] and [Lodash docs]
